### ISSUE TITLE
Code Review

### ISSUE THREAD

**Issue by m-mandel** [atefsaw/WhistleBlower#13]
Nice to see you use pull requests.
Overall, it looks like your code can improve by splitting long functions into smaller functions.
Try to make your code as understandable as possible - in such a way that if you look at your code a year from now it will be relatively easy to understand and debug. The way to improve your code is by reviewing each other's code and giving each other concrete feedback on what would make your peer's code better.

I went over some of your code, and here are some of my comments:

## ChatActivity.java

```
 protected void onCreate(Bundle savedInstanceState) {
...
}
```
This function is too long, should be split into smaller functions.

```
// this is where the message text goes
        editText = (EditText) this.findViewById(R.id.edittext_chatbox);
```
This comment is not very clear, what does message text mean?
Maybe the comment can become redundant if there is better naming of the variable.

I would maybe think of organising the function differently - instead of getting all the different strings/intents at once and then doing something with each of them, for each sub-task create it's own function. For instance, put these two lines together:
```
String groupName = getIntent().getStringExtra(getString(R.string.groupNameIntentKey));
this.setTitle(groupName);
```
and call this function `setActionBarTitle` or something similar.
This will make your code much more readable and easier to follow.

## CreateGroup.java

Name of class should end with 'Activity' if it inherits from activity - this will make the code easier to follow.
`onCreate` and `createGroup` functions are too long, you should split them into smaller functions.

`ArrayList<String> numbers;` - are these phone numbers? if so - a better name would be phoneNumbers.

## GroupListAdapter.java
```
public GroupViewHolder(View itemView, final OnItemClickListener listener) {
...
}
```
This could probably be shorter.

## GroupsActivity.java
The functions in this class can probably be shorter.
Overall, the code was hard to follow. Think how long it will take you to understand what this code does if you go back to it in a year from now. 
For example:
```
timerHandler = new Handler();
        timerRunnable = new Runnable() {
            @Override
            public void run() {
                long millis = System.currentTimeMillis();
                groups = RestHandler.pullGroups(currentUser.getUserId());
                allGroups.addAll(groups); // add the new groups to all groups
                for (Group group : groups) {
                        String groupName = group.getName();
                        String lastSeenMessage = "";
                        GroupItem item = new GroupItem(R.drawable.ic_android, groupName,
                                                       lastSeenMessage, group.getId());
                        groupsItems.add(item);
                        recyclerView.setAdapter(adapter);
                        recyclerView.setLayoutManager(layoutManager);
                }

                timerHandler.postDelayed(this, GROUP_POLLING_INTERVAL);
                }
        };
timerHandler.postDelayed(timerRunnable, 0); // Run the above polling thread immediately
```
What does this code do? Why do you need to do it? Explaining the code by putting it into its own function, and using better naming can help.

## BusinessLogic.java

```
   private void createDefaultGroup() {
        User user1 = new User("1");
        User user2 = new User("2");
        users.add(user1);
        users.add(user2);
        List<String> userIdsList = new ArrayList<>();
        userIdsList.add("1");
        userIdsList.add("2");
        createGroup(new Group(userIdsList, "First"));
    }
```
Is this code meant to be temporary or part of production? If temporary - it should not be in the master branch and with a TODO comment to remove. If part of production - there is probably a better way to do this than adding hard-coded details. For example, have the code read from some text file the details of the default group - that way you don't need to change the code in the future (if you want to change the details of the group for testing purposes for instance).

**Comment by atefsawaed** [comment]
@mosem, Thank you for the tips and notes.
We will consider fixing the things you mentioned and try to avoid them in the future.

Regards,
WhistleBlower team.
